### ISSUE TITLE
Hide transient during command execution

### ISSUE THREAD

**Issue by t-e-r-m** [magit/transient#152]
Hi,
Being able to hide the Transient during command execution would be pretty super.  Using `:transient nil` does work but it does not restore the transient.

**Comment by tarsius** [comment]
Please explain in more detail what you mean by "hide the Transient during command execution". I think I know what you mean, but would rather not speculate.

**Comment by t-e-r-m** [comment]
Without the Transient:

![2021-09-23_10:20:01_1920x1080](https://user-images.githubusercontent.com/25481939/134456127-d0cab34a-ec66-4f50-95c7-a3c7c3cd5c74.jpg)

Transient invoked:

![2021-09-23_10:20:17_1920x1080](https://user-images.githubusercontent.com/25481939/134456178-c2c58fc5-8645-49d4-b3d5-472f1e1593bd.jpg)

`find-file` called, and now very little of the original buffer is visible!

![2021-09-23_10:20:20_1920x1080](https://user-images.githubusercontent.com/25481939/134456248-334c2f63-e843-4ef8-bc66-c13d9634d29b.jpg)

What I'd like it to look like after find-file is called from the transient

![2021-09-23_10:20:24_1920x1080](https://user-images.githubusercontent.com/25481939/134456332-2a11fff8-f88d-4c95-8cd6-7260728f18c0.jpg)

The behavior in the picture above is achievable with `:transient nil`, as shown below:

```lisp
(transient-define-prefix toggle-modes-transient-c ()
  :transient-suffix     'transient--do-stay
  :transient-non-suffix 'transient--do-warn
  ["Major Modes"
   [("1" "Fundamental Mode" fundamental-mode)]]
  ["Other"
   [("B" "Revert Buffer" revert-buffer)
    ("!" (lambda () (toggle-modes-transient--description 'custom-transparency "Transparency"))
     toggle-transparency-c)]
   [("@" "Ref. Proj. cache" projectile-invalidate-cache)
    ("#" "Open Terminal here" open-external-term-here-c)
    ("F" "Find File" find-file :transient nil)]])
```
But the transient isn't restored after, because it's been exited.  I'd like it to still be active but not be visible until `find-file` finishes.

**Comment by tarsius** [comment]
Implemented in 5762bd9a06a87661e5bc6ed2dcb2d075b01e6df1.

**Comment by t-e-r-m** [comment]
Awesome!  Thanks a lot!!
